<a href="https://colab.research.google.com/github/tjddbs0401/PrimeNumber/blob/master/Correlation_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd
xls = pd.read_excel("test_project.xlsx", index_col=0)

In [0]:
xls

출생아수  평균연령  ...  인구 십만명당 사회복지시설수 (개)         범죄율
강원 강릉시    1236    43  ...                40.68   27.028059
강원 동해시     662    43  ...                11.79   26.407303
강원 삼척시     381    45  ...                10.06   21.167579
강원 속초시     517    43  ...                22.01   20.582033
강원 원주시    2468    40  ...                22.49   22.681632
강원 춘천시    1993    41  ...                31.35   25.288919
강원 태백시     244    45  ...                 8.50   30.845347
경기 고양시    7507    40  ...                18.18   33.128891
경기 과천시     414    41  ...                 4.70   29.664186
경기 광주시    3044    39  ...                14.95   28.423504
경기 구리시    1340    40  ...                10.84   24.144922
경기 군포시    2887    39  ...                16.15   34.527936
경기 김포시    3346    39  ...                14.31   32.148872
경기 남양주시   5218    39  ...                20.24   36.129972
경기 동두천시    714    42  ...                38.67   27.730530
경기 부천시    6426    40  ...                17.62   23.593083
경기 성남시    8312    40  ...                 8.72   25.893512
경기 수원시   10940    38  ...                 8.79   23.538105
경기 시흥시    3069    38  ...                21.59   17.117947
경기 안성시    1400    41  ...                36.09   25.088615
경기 안양시    4776    40  ...                 8.70   30.259535
경기 양주시    1525    40  ...                31.14   28.125496
경기 여주시     781    43  ...                36.75   22.201592
경기 오산시    2261    36  ...                 7.67   23.523788
경기 용인시    8288    38  ...                12.71   39.402322
경기 의왕시    1218    40  ...                13.40   40.423672
경기 의정부시   2967    41  ...                23.04   29.482047
경기 이천시    1879    39  ...                16.64   25.247120
경기 파주시    3876    39  ...                23.68   28.372588
경기 평택시    4179    39  ...                10.19   18.667512
...        ...   ...  ...                  ...         ...
경북 포항시    4088    42  ...                 8.13   28.879792
광주광역시    11580    39  ...                10.35   27.063827
대구광역시    18298    41  ...                14.65   28.858654
대전광역시    12436    39  ...                11.42   30.884710
부산광역시    24906    43  ...                 5.72   26.115052
서울특별시    75536    41  ...                 7.96   29.043258
세종시       3297    36  ...                 8.64   52.145033
울산광역시    10910    39  ...                 7.42   29.067791
인천광역시    23609    40  ...                16.79   28.446718
전남 광양시    1447    39  ...                14.14   31.822459
전남 나주시     902    45  ...                21.08  142.590164
전남 목표시    1823    40  ...                14.30   20.286629
전남 순천시    1961    40  ...                12.57   27.370345
전남 여수시    2286    43  ...                14.53   21.996346
전북 군산시    2104    41  ...                15.13   25.552477
전북 김제시     454    49  ...                33.04   33.984514
전북 남원시     520    47  ...                21.38   36.924561
전북 익산시    2010    42  ...                22.63   30.856336
전북 전주시    4797    40  ...                 9.05   34.416433
전북 정읍시     646    47  ...                20.84   37.128627
충남 계룡시     371    38  ...                23.46   86.303644
충남 공주시     547    45  ...                29.11   33.752226
충남 논산시     713    45  ...                30.03   33.157427
충남 당진시    1717    41  ...                16.20   20.520936
충남 보령시     627    46  ...                17.33   26.498214
충남 서산시    1564    42  ...                16.39   33.156280
충남 아산시    3441    37  ...                15.52   35.224302
충남 천안시    6021    37  ...                14.89   25.749198
충북 제천시     891    43  ...                27.84   31.261049
충북 충주시    1428    43  ...                25.92   24.988007

[80 rows x 52 columns]

In [0]:
corr_matrix = xls.corr()

In [0]:
corr_matrix["범죄율"].sort_values(ascending=False)

범죄율                       1.000000
노인 천명당 노인여가복지시설수 (개)      0.337681
합계출산율                     0.286010
인구 천명당 의료기관병상수 (개)        0.247788
인구증가율 (%)                 0.233469
독거노인가구비율 (%)              0.215378
논경지면적 (헥타르)               0.193084
고령인구비율 (%)                0.182359
인구 십만명당 문화기반시설수 (개)       0.166790
1인당 자동차 등록대수 (대)          0.130790
평균연령                      0.129331
주민 만명당 화재발생건수 (건)         0.117715
인구 십만명당 사회복지시설수 (개)       0.058538
스트레스 인지율 (%)              0.041981
비만율 (%)                   0.038753
인구십만명당 자살률 (십만명당)        -0.001621
재정자주도 (%)                -0.035380
대학교 수 (개)                -0.042108
요양기관수 (개)                -0.054494
대학교 학생수 (명)              -0.054585
조혼인율                     -0.063111
주민등록인구 (명)               -0.064495
화재발생건수 (건)               -0.065606
가구수                      -0.065767
건강보험 적용인구 현황             -0.066030
출생아수                     -0.066745
사망자수 (명)                 -0.066771
주택수 (호)                  -0.067621
사업체수 (개)            

범죄율                       1.000000
노인 천명당 노인여가복지시설수 (개)      0.337681
합계출산율                     0.286010
인구 천명당 의료기관병상수 (개)        0.247788
인구증가율 (%)                 0.233469
독거노인가구비율 (%)              0.215378
논경지면적 (헥타르)               0.193084
고령인구비율 (%)                0.182359
인구 십만명당 문화기반시설수 (개)       0.166790
1인당 자동차 등록대수 (대)          0.130790
평균연령                      0.129331
주민 만명당 화재발생건수 (건)         0.117715
인구 십만명당 사회복지시설수 (개)       0.058538
스트레스 인지율 (%)              0.041981
비만율 (%)                   0.038753
인구십만명당 자살률 (십만명당)        -0.001621
재정자주도 (%)                -0.035380
대학교 수 (개)                -0.042108
요양기관수 (개)                -0.054494
대학교 학생수 (명)              -0.054585
조혼인율                     -0.063111
주민등록인구 (명)               -0.064495
화재발생건수 (건)               -0.065606
가구수                      -0.065767
건강보험 적용인구 현황             -0.066030
출생아수                     -0.066745
사망자수 (명)                 -0.066771
주택수 (호)                  -0.067621
사업체수 (개)                 -0.068283
초등학교 교원수 (명)             -0.070791
등록외국인 현황  (명)            -0.074406
초등학교 학생수 (명)             -0.074623
일반회계중 일반공공행정예산비중 (%)     -0.077395
초등학교수 (개)                -0.081853
유치원 원아수 (명)              -0.086653
유치원수 (개)                 -0.094389
폐수배출업소수 (개소)             -0.120295
유아 천명당 보육시설수 (개)         -0.124157
교원1인당 학생수 (명)            -0.128819
고위험음주율 (%)               -0.136667
음주율 (%)                  -0.138396
재정자립도 (%)                -0.160542
인구 천명당 외국인수 (명)          -0.175310
남녀성비 (%)                 -0.184414
하수도보급률 (%)               -0.193198
자동차 천대당 교통사고발생건수 (건)     -0.207803
주민 1인당 생활폐기물배출량 (kg/일)   -0.228728
인구 천명당 사설학원수 (개)         -0.231967
흡연율 (%)                  -0.243059
상수도보급률 (%)               -0.243113
일반회계중 사회복지예산비중 (%)       -0.253021
조이혼율                     -0.292742